File: Data Structures & Algorithms/largest-rectangle-in-histogram/submission-0.py
class Solution:
    def largestRectangleArea(self, heights: List[int]) -> int:

        res = -1


        for i in range(len(heights)):
            Min_ht = heights[i]
            for j in range(i,len(heights)):
                
                Min_ht = min(Min_ht,heights[j])
                res = max(res,(j-i+1)*Min_ht)

        return res


        

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int n=heights.length;
        int i=0;
        int j=n-1;
        int res=0;
        while(i<j)
        {
            int water=(j-i)*Math.min(heights[i],heights[j]);
            res=Math.max(res,water);
            if(heights[i]<heights[j]) i++;
            else
             j--;
        }
        return res;
    }
}
